[markup] NavButton, NavMenu, Navigation, Pagenation 컴포넌트 마크업

jump6746 opened the pull request:
## #️⃣ Issue Number

ex) #이슈번호, #이슈번호
close #39 
close #43 
close #8
close #19
## 📝 요약(Summary)

변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요.
### _sprite.svg
- 최신화

### NavButton
- 하위메뉴가 있다면 button 태그 없다면 Link 태그가 될 수 있게 구현
- Link태그의 주소를 받을 수 있게 props에 link를 추가했습니다.
- 하위메뉴 드롭다운 할 수 있게 애니메이션 넣었습니다.

### NavMenu
- 컴포넌트에 맞게 link 속성 추가했습니다.
- children 여부에 따라 isActive 다르게 동작하게 설정했습니다.
- 해당 주소가 열려있다면 드롭다운이 안 닫히게 설정했습니다.
- 주소가 바뀌어도 드롭다운은 열려있습니다.

### Navigation
- Navigation 구현했습니다.

### Pagenation
- Pagenation 기본 단위 5개로 구현
- 단위는 언제든 props로 받게 변경할 수 있게 변수로 뽑았습니다.
- disabled일때 배경색만 회색으로 변경

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 💬 공유사항 to 리뷰어

Pagenation에서 버튼을 일단 disabled일때 회색으로 변경하게 했는데
아이콘 색깔까지도 변경해야할까요?

ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

## 📸스크린샷 (선택)

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

jump6746 committed:
[markup] NavButton 컴포넌트 제작
- NavButton 컴포넌트 및 스토리북 제작
- 아직 NavButton의 props에 items 연결안함

-

jump6746 committed:
[markup] : NavButton 컴포넌트 및 스토리북 제작
- props에 Link 태그를 위한 link 추가했습니다.
- hasSub 유무로 Link 태그가 되거나 Button 태그가 됩니다.
- 아직 items는 연결하지 않았습니다.

-

jump6746 committed:
Merge branch 'develop' of https://github.com/jump6746/honeyboard-client into feat/jump6746

jump6746 committed:
[markup] Pagenation 컴포넌트 및 스토리북 제작
- Pagenation 컴포넌트 및 스토리북 제작
- _sprite.svg 수정
- Pagenation 에서 버튼 disabled 배경색만 회색으로 설정
- 페이지네이션 단위를 props로 받게 된다면 쉽게 할 수 있게 설정

-

jump6746 committed:
[refactor] index.ts에 컴포넌트 등록
- index.ts에 컴포넌트 등록

-

clapsheep commented on src/components/atoms/Pagenation/Pagenation.tsx:
```diff
@@ -0,0 +1,73 @@
+interface PagenationProps {
+    total: number;
+    now: number;
+    onClickLeft: (e: React.MouseEvent) => void;
```
(e: React.MouseEvent\<HTMLButtonElement\>) => void;
이게 아마 마크다운이라 이슈 문서에서 안보였던 것 같아요 ㅠㅜ 제네릭까지 명확하게 명시해주세요!

clapsheep commented on src/components/atoms/NavButton/NavButton.tsx:
```diff
@@ -0,0 +1,66 @@
+    link?: string;
+    // items: SubNavButtonProps[];
+    isActive?: boolean;
+    onClick?: (e: React.MouseEvent) => void;
```
아래 언급한 부분이랑 동일한 이슈에요!

clapsheep reviewed (commented):
말씀드린 부분만 수정부탁드려요!

antdundun commented on src/components/atoms/NavButton/NavButton.stories.tsx:
```diff
@@ -0,0 +1,48 @@
+        hasSub: false,
+        isActive: false,
+        link: '/main',
+        onClick: () => {
```
NoSub랑 HasSub랑 onClick이 바뀐 것 같습니다!

clapsheep replied:
이게 나는 sub에 대해서 prop으로 있냐 없냐를 받아야할까 의문인게, 밑에 배열로 메뉴트리를 잘 구현해놧잖아? 거기서 children이 이있으면 없으면으로 조건부렌더링을 할 수 있지 않을까? 괜히 같은건데 prop 만 많아지는 느낌..?

jump6746 replied:
수정하겠습니다

clapsheep committed:
Merge branch 'develop' into feat/jump6746

jump6746 committed:
Merge branch 'develop' of https://github.com/jump6746/honeyboard-client into feat/jump6746

jump6746 committed:
[refactor] NavButton, Pagenation 컴포넌트 수정
- 컴포넌트의 onClick 타입 수정
- NavButton에 하위 메뉴 동작 연결

-

jump6746 committed:
[refactor] 아이콘 추가 및 _sprite.svg 생성
- 빨간 x 동그라미 아이콘 추가

-

jump6746 committed:
[refactor] SubNavButton 컴포넌트 수정
- SubNavButton 버튼 태그에서 Link 태그로 수정
- props에 onClick 대신 link 추가

-

jump6746 committed:
[markup] NavMenu 컴포넌트 제작
- NavMenu 컴포넌트 및 스토리북 구현
- NavMenu의 Props를 자식들에게 맞게 icon 추가
- Children 요소까지 isActive 받도록 제작

-

jump6746 committed:
[refactor] Persona, NavMenu 컴포넌트 스타일 추가
- Persona 패딩 추가
- NavMenu 하단 마진 추가

-

jump6746 committed:
[markup] Navigation 컴포넌트 구현
- Navigation 컴포넌트, 스토리북 제작 및 구현
closes #43

-

jump6746 committed:
Merge branch 'feat/jump6746' of https://github.com/jump6746/honeyboard-client into feat/jump6746

jump6746 committed:
[refactor] NavButton 컴포넌트 수정
- NavButton의 hasSub props를 제거했습니다.
- Navigation 디자인 구조 변경했습니다.

-

jump6746 committed:
[refactor] NavMenu 컴포넌트 수정
- NavMenu에서 기본 경로를 "/" 로 사용해서 서로 구별해서 작동할 수 있게 수정

-

jump6746 committed:
[refactor] NavMenu, NavButton 컴포넌트 수정
- NavButton의 active시 css 수정
- 경로에 따라 토글이 닫힐 수 있게 수정

-

jump6746 committed:
[refactor] Navigation, NavButton 수정
- NavButton을 사용하는 곳에서 className props로 넣을 수 있게 변경
- Navigation 디자인 변경

-

clapsheep committed:
Merge branch 'develop' into feat/jump6746
